part2/phonebook/src/App.js

```diff
@@ -54,31 +59,33 @@ const App = () => {
             number: newNumber
         }
 
-        const person = persons.find(p => p.name === newName)
-        
-        if (persons.some(person => person.name === newName)) {
+        const registeredPerson = persons.find(p => p.name === newName)
+        // persons.some(person => person.name === newName)
+        if (registeredPerson) {
             const updateNumber = window.confirm(`${newName} is already in phonebook. Replace numbers?`)
-            
+
             if (updateNumber) {
-                const updatedPerson = { ...person, number: newNumber }
+                const newPerson = { ...registeredPerson, number: newNumber }
                 personService
-                    .update(updatedPerson.id, updatedPerson)
-                    .then(() => {
-                        setPersons(persons.map(p => p.id !== updatedPerson.id ? p : updatedPerson))
+                    .update(newPerson.id, registeredPerson)
+                    .then((updatedPerson) => {
```

Comment:
Še neizskatās, ka numurs tiktu atjaunots.

1. 62\. rindā tu atrodi eksistējošu personu
2. 68\. rindā tu definē jaunu personu, ņemot par pamatu eksistējošo un izmainot numuru
3. 70\. rindā tu piesaki apdeitot id no jaunās personas (kas ir eksistējošā persona), bet kā apdeitoto vērtību, tu iedod to pašu eksistējošo, kuram numurs nav nomainīts
    - Es vispār teiktu pat to loģiku apgriezt otrādi: `.update(registeredPerson.id, newPerson)` ... pat ja ir zināms, ka `newPerson.id` būs tāds pats kā `registeredPerson.id`, vieglāk kodu izprast būs, ja lietas būs īstajās vietās (tas attiecināms in general, katru reizi vērts piedomāt, kā kodu rakstīt nepārprotamu pašu par sevi)